pynetbox/models/dcim.py

```diff
@@ -154,7 +222,26 @@ def __str__(self):
 
 class Cables(Record):
     def __str__(self):
-        return "{} <> {}".format(self.termination_a, self.termination_b)
+        # populate the terminations to get the full names if they are not already
+        try:
+            termination_a_name = self.termination_a.name
+        except AttributeError:
+            try:
+                self.termination_a.full_details()
+            except TypeError:
+                self.termination_a.full_details(self)
+            termination_a_name = getattr(self.termination_a, "name", self.termination_a)
```

Comment:
I don't really like this logic but I'm not sure of a better method. Ideas welcomed

Reply:
This was a change to force each of the returned items along the trace to get fully populated so that when you printed their name in the trace it wasn't just displayed as a `Record` item with no data. Once each item was accessed they would then print their name properly, but having a trace full of unlabeled hops was not very useful. It's not very clear, and I don't like it at all. I think it may be better to do something like this in the trace function itself (I think maybe you mentioned that before?) so that only on a trace will it try to prefetch the data of each item in the hops so they print out nicely.

Reply:
That's weird. Do you have an example you can show of the old behavior? Those should be Termination objects, but either way even Record's default __str__ behavior is to return .name, iirc.

Reply:
Sure, with this section of code reverted this is what I'm getting now:

```
In [1]: iface.trace()                                                                                                             
Out[1]: 
[[em1,
  <class 'pynetbox.models.dcim.Termination'> <> <class 'pynetbox.models.dcim.Termination'>,
  pair-11 (ports 21-22)],
 [port-2,
  <class 'pynetbox.models.dcim.Termination'> <> <class 'pynetbox.models.dcim.Termination'>,
  port-2],
 [pair-11 (ports 21-22),
  <class 'pynetbox.models.dcim.Termination'> <> <class 'pynetbox.models.dcim.Termination'>,
  Ethernet11]]

```

So it looks like maybe I was mistaken or confused about only the middle (not the first or last) having the issue, it may be all of the items returned from the trace until they are accessed in some way. You can see here how the info is filled out once the item is accessed directly: 
```
In [2]: trace = iface.trace()                                                                                                     

In [3]: trace                                                                                                                     
Out[3]: 
[[em1,
  <class 'pynetbox.models.dcim.Termination'> <> <class 'pynetbox.models.dcim.Termination'>,
  pair-11 (ports 21-22)],
 [port-2,
  <class 'pynetbox.models.dcim.Termination'> <> <class 'pynetbox.models.dcim.Termination'>,
  port-2],
 [pair-11 (ports 21-22),
  <class 'pynetbox.models.dcim.Termination'> <> <class 'pynetbox.models.dcim.Termination'>,
  Ethernet11]]

In [4]: trace[0][1]                                                                                                               
Out[4]: em1 <> pair-11 (ports 21-22)

In [5]: trace                                                                                                                     
Out[5]: 
[[em1, em1 <> pair-11 (ports 21-22), pair-11 (ports 21-22)],
 [port-2,
  <class 'pynetbox.models.dcim.Termination'> <> <class 'pynetbox.models.dcim.Termination'>,
  port-2],
 [pair-11 (ports 21-22),
  <class 'pynetbox.models.dcim.Termination'> <> <class 'pynetbox.models.dcim.Termination'>,
  Ethernet11]]

```

Reply:
By moving this to inside the trace function it is muuuuch clearer now.